### ISSUE TITLE
cdktf typescript cloudflare rate limit headers is not map

### ISSUE THREAD

**Issue by ivan-georgiev-zetta** [hashicorp/terraform-cdk#2774]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### cdktf & Language Versions

typescript
"cdktf": "^0.15.0"

### Affected Provider / Resource(s)

"cloudflare/cloudflare@~> 3.14"

### Debug Output

```
Inappropriate value for attribute "headers": list of map of string required.
```

### Expected Behavior

I can apply a rate limit rule with response headers specified.

### Actual Behavior

I receive an error because headers is defined as { [key: string]: string } | cdktf.IResolvable, but it should be { [key: string]: string }[] | cdktf.IResolvable

### Steps to Reproduce

Try to create a rate limit Cloudflare rule using cdktf and typescript. Supply RateLimitMatchResponse headers.

### References

https://github.com/cdktf/cdktf-provider-cloudflare/blob/main/src/rate-limit/index.ts#L480

- #0000

**Comment by jsteinich** [comment]
This has been fixed in version 6.0.1 of the pre-built cloudflare provider.

**Comment by ivan-georgiev-zetta** [comment]
> This has been fixed in version 6.0.1 of the pre-built cloudflare provider.

Yes, thank you, I confirm setting `cdktf` to `^0.16.0` and the `cloudflare/cloudflare` provider to 4.4 fixes the issue.
